imbi/opensearch/operations_log.py

```diff
@@ -0,0 +1,103 @@
+import asyncio
+import dataclasses
+import logging
+import typing
+
+from imbi import models
+if typing.TYPE_CHECKING:
+    from imbi import app
+
+LOGGER = logging.getLogger(__name__)
+
+OPS_LOG = {
+    'id': {'type': 'integer'},
+    'recorded_at': {'type': 'date'},
+    'recorded_by': {'type': 'text'},
+    'completed_at': {'type': 'date'},
+    'project_id': {'type': 'integer'},
+    'project_name': {'type': 'text'},
+    'environment': {'type': 'text'},
+    'change_type': {'type': 'text'},
+    'description': {'type': 'text'},
+    'link': {'type': 'text'},
+    'notes': {'type': 'text'},
+    'ticket_slug': {'type': 'text'},
+    'version': {'type': 'text'}
+}
+
+
+class OperationsLogIndex:
+    """Class for interacting with the OpenSearch OperationsLog index"""
+
+    INDEX = 'operations-log'
+
+    def __init__(self, application: 'app.Application'):
+        self.application = application
+
+    async def create_index(self):
+        LOGGER.debug('Creating operations-log index in OpenSearch')
+        await self.application.opensearch.create_index(self.INDEX)
+
+    async def create_mapping(self):
+        await self.application.opensearch.create_mapping(
+            self.INDEX, await self._build_mappings())
+
+    async def delete_document(self, ops_log_id: typing.Union[int, str]):
+        await self.application.opensearch.delete_document(
+            self.INDEX, str(ops_log_id))
+
+    async def index_document(self, ops_log: models.OperationsLog):
+        await self.application.opensearch.index_document(
+            self.INDEX, str(ops_log.id), self._ops_log_to_dict(ops_log), True)
+
+    async def search(self, query: str, max_results: int = 1000) \
+            -> typing.Dict[str, typing.List[dict]]:
+        return await self.application.opensearch.search(
+            self.INDEX, query, max_results)
+
```

Comment:
Do we need to expose the searchable fields across the [`/ui/settings` endpoint](https://github.com/AWeber-Imbi/imbi-api/blob/0ec5fe9aff93bfe1998626885f06c900b412f5cc/imbi/endpoints/ui/settings.py#L38) the way that we do in [the project stuff](https://github.com/AWeber-Imbi/imbi-api/blob/0ec5fe9aff93bfe1998626885f06c900b412f5cc/imbi/opensearch/project.py#L74-L105)?

Reply:
No, projects' are like that because the fields are dynamic, ops logs aren't.